securesystemslib/signer/_sigstore_signer.py

```diff
@@ -0,0 +1,167 @@
+"""Signer implementation for project sigstore.
+
+Example:
+```python
+from sigstore.oidc import Issuer
+
+from securesystemslib.signer import SigstoreKey, SigstoreSigner
+
+# Create public key
+identity = "luk.puehringer@gmail.com"  # change, unless you know my password
+issuer = "https://github.com/login/oauth"
+public_key = SigstoreKey.from_dict(
+    "abcdefg",
+    {
+        "keytype": "sigstore-oidc",
+        "scheme": "Fulcio",
+        "keyval": {
+            "issuer": issuer,
+            "identity": identity,
+        },
+    },
+)
+
+# Create signer
+issuer = Issuer.production()
+token = issuer.identity_token()  # requires sign in with GitHub in a browser
+signer = SigstoreSigner(token, public_key)
+
+# Sign
+signature = signer.sign(b"data")
+
+# Verify
+public_key.verify_signature(signature, b"data")
+
+```
+
+"""
+
+import io
+import logging
+from typing import Any, Dict, Optional
+
+from securesystemslib.exceptions import (
+    UnsupportedLibraryError,
+    UnverifiedSignatureError,
+    VerificationError,
+)
+from securesystemslib.signer._signer import (
+    Key,
+    SecretsHandler,
+    Signature,
+    Signer,
+)
+
+IMPORT_ERROR = "sigstore library required to use 'sigstore-oidc' keys"
+
+logger = logging.getLogger(__name__)
+
+
+class SigstoreKey(Key):
+    """Sigstore verifier."""
+
+    @classmethod
+    def from_dict(cls, keyid: str, key_dict: Dict[str, Any]) -> "SigstoreKey":
+        keytype = key_dict.pop("keytype")
+        scheme = key_dict.pop("scheme")
+        keyval = key_dict.pop("keyval")
+
+        for content in ["identity", "issuer"]:
+            if content not in keyval or not isinstance(keyval[content], str):
+                raise ValueError(
+                    f"{content} string required for scheme {scheme}"
+                )
+
+        return cls(keyid, keytype, scheme, keyval, key_dict)
+
+    def to_dict(self) -> Dict:
+        return {
+            "keytype": self.keytype,
+            "scheme": self.scheme,
+            "keyval": self.keyval,
+            **self.unrecognized_fields,
+        }
+
+    def verify_signature(self, signature: Signature, data: bytes) -> None:
+        # pylint: disable=import-outside-toplevel,import-error
+        try:
+            from sigstore.verify import VerificationMaterials, Verifier
+            from sigstore.verify.policy import Identity
+            from sigstore_protobuf_specs.dev.sigstore.bundle.v1 import Bundle
+
+            verifier = Verifier.production()
```

Comment:
seeing this code it seems clear that the content in our key should include everything needed to choose the sigstore "instance".

That said, it's quite unclear to me what that all means (rekor and tuf urls? fulcio?): this is really more a question for the TAP than this implementation which should focus on supporting the default instance for now

So no action requested here.